{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0",
    "cveMetadata": {
        "cveId": "CVE-2023-37274",
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "state": "PUBLISHED",
        "assignerShortName": "GitHub_M",
        "dateReserved": "2023-06-29T19:35:26.440Z",
        "datePublished": "2023-07-13T22:33:41.290Z",
        "dateUpdated": "2023-07-13T22:33:41.290Z"
    },
    "containers": {
        "cna": {
            "title": "Python code execution sandbox escape in non-docker version in Auto-GPT",
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-94",
                            "lang": "en",
                            "description": "CWE-94: Improper Control of Generation of Code ('Code Injection')",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "HIGH",
                        "attackVector": "LOCAL",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.6,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "HIGH",
                        "integrityImpact": "HIGH",
                        "privilegesRequired": "LOW",
                        "scope": "CHANGED",
                        "userInteraction": "REQUIRED",
                        "vectorString": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:C/C:H/I:H/A:H",
                        "version": "3.1"
                    }
                }
            ],
            "references": [
                {
                    "name": "https://github.com/Significant-Gravitas/Auto-GPT/security/advisories/GHSA-5h38-mgp9-rj5f",
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/Significant-Gravitas/Auto-GPT/security/advisories/GHSA-5h38-mgp9-rj5f"
                },
                {
                    "name": "https://github.com/Significant-Gravitas/Auto-GPT/pull/4756",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/Significant-Gravitas/Auto-GPT/pull/4756"
                }
            ],
            "affected": [
                {
                    "vendor": "Significant-Gravitas",
                    "product": "Auto-GPT",
                    "versions": [
                        {
                            "version": "< 0.4.3",
                            "status": "affected"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M",
                "dateUpdated": "2023-07-13T22:33:41.290Z"
            },
            "descriptions": [
                {
                    "lang": "en",
                    "value": "Auto-GPT is an experimental open-source application showcasing the capabilities of the GPT-4 language model. When Auto-GPT is executed directly on the host system via the provided run.sh or run.bat files, custom Python code execution is sandboxed using a temporary dedicated docker container which should not have access to any files outside of the Auto-GPT workspace directory.\nBefore v0.4.3, the `execute_python_code` command (introduced in v0.4.1) does not sanitize the `basename` arg before writing LLM-supplied code to a file with an LLM-supplied name. This allows for a path traversal attack that can overwrite any .py file outside the workspace directory by specifying a `basename` such as `../../../main.py`. This can further be abused to achieve arbitrary code execution on the host running Auto-GPT by e.g. overwriting autogpt/main.py which will be executed outside of the docker environment meant to sandbox custom python code execution the next time Auto-GPT is started. The issue has been patched in version 0.4.3. As a workaround, the risk introduced by this vulnerability can be remediated by running Auto-GPT in a virtual machine, or another environment in which damage to files or corruption of the program is not a critical problem."
                }
            ],
            "source": {
                "advisory": "GHSA-5h38-mgp9-rj5f",
                "discovery": "UNKNOWN"
            }
        }
    }
}